q2_keegan/vanilla/_format.py

```diff
@@ -0,0 +1,33 @@
+import re
+
+import qiime2.plugin.model as model
+from qiime2.plugin import ValidationError
+
+from ..plugin_setup import plugin
+
+
+class VanillaBeanFmt(model.TextFileFormat):
+
+    def _validate_(self, level):
```

Comment:
Right now the `level` parameter is unused. The API for `validate()` specifies a level value, that can either be `'min'` or `'max'`. By convention the idea is that `min` should perform the "minimum" amount of validation necessary, while `max` is the "maximum." In practice, `min` usually validates the schema of the data ("does it have the right shape and characteristics ie column count, etc?") and often the first few records of the data. `max` is schema plus _all_ of the data.